wordpress-mobile/WordPress-iOS
# Use fastlane match to manage certificates and profiles for AppStore and Enterprise builds

Currently, we create and manage our certificates and provisioning profiles on an ad-hoc basis. This means that certs and profiles need to be created and renewed manually, but more importantly, they need to be downloaded and installed manually. This means there is no way for CI to easily install and use our certs/profiles.

This PR updates our internal (enterprise) and AppStore builds to use [Fastlane Match](https://docs.fastlane.tools/actions/match/) to create and manage their certs/profiles. They are [stored encrypted in Google Cloud](https://docs.fastlane.tools/actions/match/#google-cloud-storage).

This means there is no more need to renew profiles and certs manually and that they can easily be used on CI (I have a CircleCI build using it [here](https://circleci.com/gh/wordpress-mobile/WordPress-iOS/13403)).

We could extend this to also work for development certs/profiles but I think its a good idea to see how this goes first.

To test:

1. Run `bundle exec fastlane build_and_upload_release` from the `Scripts` directory.
2. It will check that certs/profiles are up to date and install them if needed.
3. See that the internal build is successfully uploaded to Hockey.
4. The AppStore build will fail at the final step because the current version already exists.
5. Delete the new build in HockeyApp.

Update release notes:
